test/test_healthcheck.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"time"
+
+	c "github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-check"
+	ct "github.com/flynn/flynn/controller/types"
+	"github.com/flynn/flynn/host/types"
+)
+
+type HealthcheckSuite struct {
+	Helper
+}
+
+var _ = c.Suite(&HealthcheckSuite{})
+
+func (s *HealthcheckSuite) createAppWithService(t *c.C, process string, service *host.Service) (*ct.App, *ct.Release) {
+	app, release := s.createApp(t)
+
+	release.ID = ""
+	proctype := release.Processes[process]
+	proctype.Ports = []ct.Port{{
+		Proto:   "tcp",
+		Service: service,
+	}}
+	release.Processes[process] = proctype
+
+	t.Assert(s.controllerClient(t).CreateRelease(release), c.IsNil)
+	t.Assert(s.controllerClient(t).SetAppRelease(app.ID, release.ID), c.IsNil)
+	return app, release
+}
+
+func (s *HealthcheckSuite) TestChecker(t *c.C) {
+	// start app with ping service, register with checker
+	app, _ := s.createAppWithService(t, "ping", &host.Service{
+		Name:   "ping-checker",
+		Create: true,
+		Check:  &host.HealthCheck{Type: "tcp"},
+	})
+	t.Assert(flynn(t, "/", "-a", app.Name, "scale", "ping=1"), Succeeds)
+	_, err := s.discoverdClient(t).Instances("ping-checker", 10*time.Second)
+	t.Assert(err, c.IsNil)
+}
+
+func (s *HealthcheckSuite) TestWithoutChecker(t *c.C) {
+	// start app with a service but no checker
+	app, _ := s.createAppWithService(t, "ping", &host.Service{
+		Name:   "ping-without-checker",
+		Create: true,
+	})
+	t.Assert(flynn(t, "/", "-a", app.Name, "scale", "ping=1"), Succeeds)
+	// make sure app is registered and unregistered when the process terminates
+	_, err := s.discoverdClient(t).Instances("ping-without-checker", 10*time.Second)
```

Comment:
This doesn't do what it says on the tin. Where is the verification that it is unregistered?